### ISSUE TITLE
Help wanted on the "split" schedule

### ISSUE THREAD

**Issue by roastduck** [tensor-compiler/taco#336]
I tried to compile the following schedule with the command line tool:

```sh
../build/bin/taco "C(i, j) = A(i, k) * B(k, j)" -s="split(i, i0, i1, 32)"
```

It failed with the message "The index statement is not valid concrete index notation: all variables in concrete notation must be bound by a forall statement". What does that mean? What is a forall statement?

However, if I change `i` to be `j`, the following schedule works fine and outputs the target code:

```sh
../build/bin/taco "C(i, j) = A(i, k) * B(k, j)" -s="split(j, j0, j1, 32)"
```

Are there any prerequisites to use the schedule?

**Comment by RSenApps** [comment]
Unfortunately I wasn't able to reproduce the issue:
```c
➜  build git:(master) bin/taco "C(i, j) = A(i, k) * B(k, j)" -s="split(i, i0, i1, 32)"
// Generated by the Tensor Algebra Compiler (tensor-compiler.org)

int compute(taco_tensor_t *C, taco_tensor_t *A, taco_tensor_t *B) {
  int C1_dimension = (int)(C->dimensions[0]);
  int C2_dimension = (int)(C->dimensions[1]);
  double* restrict C_vals = (double*)(C->vals);
  int A1_dimension = (int)(A->dimensions[0]);
  int A2_dimension = (int)(A->dimensions[1]);
  double* restrict A_vals = (double*)(A->vals);
  int B1_dimension = (int)(B->dimensions[0]);
  int B2_dimension = (int)(B->dimensions[1]);
  double* restrict B_vals = (double*)(B->vals);

  #pragma omp parallel for schedule(static)
  for (int32_t pC = 0; pC < (C1_dimension * C2_dimension); pC++) {
    C_vals[pC] = 0.0;
  }

  for (int32_t i0 = 0; i0 < ((A1_dimension + 31) / 32); i0++) {
    for (int32_t i1 = 0; i1 < 32; i1++) {
      int32_t i = i0 * 32 + i1;
      if (i >= A1_dimension)
        continue;

      for (int32_t k = 0; k < B1_dimension; k++) {
        int32_t kA = i * A2_dimension + k;
        for (int32_t j = 0; j < B2_dimension; j++) {
          int32_t jC = i * C2_dimension + j;
          int32_t jB = k * B2_dimension + j;
          C_vals[jC] = C_vals[jC] + A_vals[kA] * B_vals[jB];
        }
      }
    }
  }
  return 0;
}
```

This is on Ubuntu 18.04 with gcc 7.5.0 and the latest commit hash 0a62520f72b210fefc7978fc9c36aecea84bd8d9. Could you provide similar instructions for how to reproduce your build of taco? You can also try to use the web demo in the meantime http://tensor-compiler.org/codegen.html

**Comment by roastduck** [comment]
I found that I failed only with `-DCUDA=ON` option on for cmake. I can run without problems after turning this option off.

Tested with Debian 10, GCC 8.3.0, CUDA 10.2, at commit 0a62520.

**Comment by roastduck** [comment]
Ping @RSenApps. Can you reproduce the issue with `-DCUDA=ON`?

**Comment by RSenApps** [comment]
Thanks for the information and reporting the issue, I was able to reproduce and fix with commit 486cd9f464e2d8f603cf3b3caf749c6389d378f4
